src/layout/layout.rs

```diff
@@ -1842,237 +1843,88 @@ mod tests {
-            assert_eq!([[a.x, a.width], [b.x, b.width]], [[0, 50], [50, 50]]);
-
-            // min should justify to start
-            let [a, b] = Rect::new(0, 0, 100, 1)
-                .split(&Layout::horizontal([Min(25), Min(25)]).flex(Flex::Start));
-            assert_eq!([[a.x, a.width], [b.x, b.width]], [[0, 25], [25, 25]]);
-
-            // min should justify to the center
-            let [a, b] = Rect::new(0, 0, 100, 1)
-                .split(&Layout::horizontal([Min(25), Min(25)]).flex(Flex::Center));
-            assert_eq!([[a.x, a.width], [b.x, b.width]], [[25, 25], [50, 25]]);
-
-            // min should justify to the end
-            let [a, b] = Rect::new(0, 0, 100, 1)
-                .split(&Layout::horizontal([Min(25), Min(25)]).flex(Flex::End));
-            assert_eq!([[a.x, a.width], [b.x, b.width]], [[50, 25], [75, 25]]);
+        #[rstest]
+        #[case::length_stretches_to_end([Length(50)], Flex::StretchLast, [0, 100])]
+        #[case::length_stretches([Length(50)], Flex::Stretch, [0, 100])]
+        #[case::length_left_justified([Length(50)], Flex::Start, [0, 50])]
+        #[case::length_right_justified([Length(50)], Flex::End, [50, 50])]
+        #[case::length_center_justified([Length(50)], Flex::Center, [25, 50])]
+        #[case::fixed_stretches_to_end([Fixed(50)], Flex::StretchLast, [0, 100])]
+        #[case::fixed_left_justified([Fixed(50)], Flex::Start, [0, 50])]
+        #[case::fixed_right_justified([Fixed(50)], Flex::End, [50, 50])]
+        #[case::fixed_center_justified([Fixed(50)], Flex::Center, [25, 50])]
+        #[case::ratio_stretches_to_end([Ratio(1, 2)], Flex::StretchLast, [0, 100])]
+        #[case::ratio_left_justified([Ratio(1, 2)], Flex::Start, [0, 50])]
+        #[case::ratio_right_justified([Ratio(1, 2)], Flex::End, [50, 50])]
+        #[case::ratio_center_justified([Ratio(1, 2)], Flex::Center, [25, 50])]
+        #[case::percent_stretches_to_end([Percentage(50)], Flex::StretchLast, [0, 100])]
+        #[case::percent_left_justified([Percentage(50)], Flex::Start, [0, 50])]
+        #[case::percent_right_justified([Percentage(50)], Flex::End, [50, 50])]
+        #[case::percent_center_justified([Percentage(50)], Flex::Center, [25, 50])]
+        #[case::min_stretches_to_end([Min(50)], Flex::StretchLast, [0, 100])]
+        #[case::min_left_justified([Min(50)], Flex::Start, [0, 50])]
+        #[case::min_right_justified([Min(50)], Flex::End, [50, 50])]
+        #[case::min_center_justified([Min(50)], Flex::Center, [25, 50])]
+        #[case::max_stretches_to_end([Max(50)], Flex::StretchLast, [0, 100])]
+        #[case::max_left_justified([Max(50)], Flex::Start, [0, 50])]
+        #[case::max_right_justified([Max(50)], Flex::End, [50, 50])]
+        #[case::max_center_justified([Max(50)], Flex::Center, [25, 50])]
+        fn flex_one_constraint(
+            #[case] constraints: [Constraint; 1],
+            #[case] flex: Flex,
+            #[case] expected_widths: [u16; 2],
```

Comment:
Perhaps make this either a tuple or multiple values - (start, width)

Reply:
Changed everything to tuples.